feat(optimize): add optimize example project (#6444)
prisma/prisma-examples: optimize/starter/README.md

Hey there!

Welcome to Yet Another Todo App: Optimize Edition!

To get started, follow these steps:

> 1. Run `npm i` from project root.
> 1. Login and create yourself an API key (also called token) [in the Optimize dashboard](https://optimize.prisma.io).
> 1. Navigate to `packages/server` and add a `.env` file that matches the `.env.example` file.

Reviewer: ```suggestion
1. Run `npm i` from project root.
2. Login and create yourself an API key (also called token) [in the Optimize dashboard](https://optimize.prisma.io).
3. Navigate to `packages/server` and add a `.env` file that matches the `.env.example` file.
```

Author: This is just md shorthand. It changes itself to be properly numbered.

<img width="1159" alt="image" src="https://github.com/user-attachments/assets/516ffecd-4d2c-44a1-8899-821915d78f82">

   ```sh
   cd packages/server
   cp .env.example .env
   ```

> 1. Copy your API key (also called token) to the `PRISMA_OPTIMIZE_TOKEN`.

Reviewer: ```suggestion
4. Copy your API key (also called token) to the `PRISMA_OPTIMIZE_TOKEN`.
```

> 1. Run `npm i` from project root.
> 1. Login and create yourself an API key (also called token) [in the Optimize dashboard](https://optimize.prisma.io).
> 1. Navigate to `packages/server` and add a `.env` file that matches the `.env.example` file.
>    ```sh
>    cd packages/server
>    cp .env.example .env
>    ```
> 1. Copy your API key (also called token) to the `PRISMA_OPTIMIZE_TOKEN`.
> 
>    ```
>    PRISMA_OPTIMIZE_TOKEN="<your_token_here>"
>    ```
> 
> 1. Run `npm start` from project root.
> 1. Go back to recordings [in the Optimize dashboard](https://optimize.prisma.io) and click the "Start recording" button.
> 1. Open the demo app @ http://localhost:5174 in the browser. Perform a bunch of actions - add a Todo, mark a Todo as complete, delete a Todo, etc.
> 1. Go back to the Optimize dashboard and click the "Stop recording" button.
> 1. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly _Ask Justin_ AI chat.

Reviewer: ```suggestion
1. Run `npm i` from project root.
2. Login and create yourself an API key (also called token) [in the Optimize dashboard](https://optimize.prisma.io).
3. Navigate to `packages/server` and add a `.env` file that matches the `.env.example` file.
   ```sh
   cd packages/server
   cp .env.example .env
   ```
4. Copy your API key (also called token) to the `PRISMA_OPTIMIZE_TOKEN`.

   ```
   PRISMA_OPTIMIZE_TOKEN="<your_token_here>"
   ```

5. Run `npm start` from project root.
6. Go back to recordings [in the Optimize dashboard](https://optimize.prisma.io) and click the "Start recording" button.
7. Open the demo app @ http://localhost:5174 in the browser. Perform a bunch of actions - add a Todo, mark a Todo as complete, delete a Todo, etc.
8. Go back to the Optimize dashboard and click the "Stop recording" button.
9. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly AI chat.
```

> 1. Run `npm start` from project root.
> 1. Go back to recordings [in the Optimize dashboard](https://optimize.prisma.io) and click the "Start recording" button.
> 1. Open the demo app @ http://localhost:5174 in the browser. Perform a bunch of actions - add a Todo, mark a Todo as complete, delete a Todo, etc.
> 1. Go back to the Optimize dashboard and click the "Stop recording" button.
> 1. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly _Ask Justin_ AI chat.

Reviewer: ```suggestion
5. Run `npm start` from project root.
6. Go back to recordings [in the Optimize dashboard](https://optimize.prisma.io) and click the "Start recording" button.
7. Open the demo app @ http://localhost:5174 in the browser. Perform a bunch of actions - add a Todo, mark a Todo as complete, delete a Todo, etc.
8. Go back to the Optimize dashboard and click the "Stop recording" button.
9. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly AI chat.
```

Hey there!

Welcome to Yet Another Todo App: Optimize Edition!

To get started, follow these steps:

1. Run `npm i` from project root.
1. Login and create yourself an API key (also called token) [in the Optimize dashboard](https://optimize.prisma.io).
1. Navigate to `packages/server` and add a `.env` file that matches the `.env.example` file.
   ```sh
   cd packages/server
   cp .env.example .env
   ```
1. Copy your API key (also called token) to the `PRISMA_OPTIMIZE_TOKEN`.

   ```
   PRISMA_OPTIMIZE_TOKEN="<your_token_here>"
   ```

1. Run `npm run build` from project root to create your SQLite database.
1. Go back to recordings [in the Optimize dashboard](https://optimize.prisma.io) and click the "Start recording" button.
1. Run `npm start` from project root.
1. Open the demo app @ http://localhost:5174 in the browser. Perform a bunch of actions - add a Todo, mark a Todo as complete, delete a Todo, etc.
1. Go back to the Optimize dashboard and click the "Stop recording" button.

> 1. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly _Ask Justin_ AI chat.

Reviewer: ```suggestion
1. Enjoy some sweet queries, groups, and most importantly recommendations with our friendly AI chat.
```
